backend/src/main/kotlin/no/uib/echo/routes/RegistrationRoutes.kt

```diff
@@ -297,32 +314,46 @@ fun Route.postRegistration(sendGridApiKey: String?, sendEmail: Boolean) {
+                if (oldReg[Registration.registrationStatus] != Status.DEREGISTERED) {
+
+                    val responseCode = when (oldReg[Registration.registrationStatus]) {
+                        Status.WAITLIST -> RegistrationResponse.AlreadySubmittedWaitList
+                        else -> RegistrationResponse.AlreadySubmitted
+                    }
+                    call.respond(
+                        HttpStatusCode.UnprocessableEntity,
+                        resToJson(responseCode)
+                    )
+                    return@post
+                }
+                transaction {
+                    Registration.update({ Registration.userEmail eq registration.email.lowercase() and (Registration.happeningSlug eq registration.slug) }) {
+                        it[degree] = userDegree.toString()
+                        it[degreeYear] = userDegreeYear
+                        it[registrationStatus] = if (waitList) Status.WAITLIST else Status.REGISTERED
+                    }
+                }
+            } else {
+
+                transaction {
+                    Registration.insert {
+                        it[userEmail] = registration.email.lowercase()
+                        it[happeningSlug] = registration.slug
+                        it[degree] = userDegree.toString()
+                        it[degreeYear] = userDegreeYear
+                        it[registrationStatus] = if (waitList) Status.WAITLIST else Status.REGISTERED
+                    }
                 }
-                call.respond(
-                    HttpStatusCode.UnprocessableEntity, resToJson(responseCode)
-                )
-                return@post
             }
 
             transaction {
-                Registration.insert {
-                    it[userEmail] = registration.email.lowercase()
-                    it[happeningSlug] = registration.slug
-                    it[degree] = userDegree.toString()
-                    it[degreeYear] = userDegreeYear
-                    it[Registration.waitList] = waitList
-                }
+                Answer.deleteWhere { registrationEmail eq registration.email and (happeningSlug eq registration.slug) }
 
-                if (registration.answers.isNotEmpty()) {
-                    Answer.batchInsert(registration.answers) { a ->
-                        this[Answer.registrationEmail] = registration.email.lowercase()
-                        this[Answer.happeningSlug] = registration.slug
-                        this[Answer.question] = a.question
-                        this[Answer.answer] = a.answer
-                    }
+                Answer.batchInsert(registration.answers) { a ->
+                    this[Answer.registrationEmail] = registration.email.lowercase()
+                    this[Answer.happeningSlug] = registration.slug
+                    this[Answer.question] = a.question
+                    this[Answer.answer] = a.answer
                 }
```

Comment:
Hva skjer her egt? Slettes svar uansett eller er det bare når man allerede er påmeldt? Og hva skjedde med sjekk for `registration.answers.isNotEmpty()`?